platform/android/java/build.gradle

```diff
@@ -29,8 +29,13 @@ allprojects {
 
 ext {
     supportedAbis = ["arm32", "arm64", "x86_32", "x86_64"]
-    supportedTargetsMap = [release: "release", dev: "debug", debug: "release_debug"]
     supportedFlavors = ["editor", "template"]
+    supportedFlavorsBuildTypes = [
+        // The editor can't be used with target=release as debugging tools are then not
+        // included, and it would crash on errors instead of reporting them.
+        "editor": ["dev", "debug"],
+        "template": ["dev", "debug", "release"]
+    ]
```

Comment:
I don't know what gradle needs but I think it would be clearer if we followed the same structure as in scons, so:
```
supportedFlavors = ["editor", "template_debug", "template_release"]
supportedBuildTypes = ["regular", "dev"]  // In SCons it's just a boolean, so could be the same here if that works.
```

`dev_build=yes` can now be used with any target so `target=template_release dev_build=yes` is also a valid use case (gives `template_release.dev` lib). It's not a particularly _useful_ combination but it's IMO better if gradle follows the same logic of having only three targets and then a bool for `dev_build`.

Reply:
The build types are also shared across the project modules, while the flavors are linked:
- The `app` module maps to the `template` flavor
- The `editor` module maps to the `editor` flavor
- As a dependency, the `lib` module is configured to automatically update its flavor and build type to match.

Reply:
Sounds great, thanks for the explanation :)